src/cli.rs

```diff
@@ -86,12 +89,18 @@ impl RunCommand for DataCommand {
     async fn run(&self, config: Config) -> Result<()> {
         info!("Running `data` subcommand");
 
+        let mut sp = Spinner::new(
+            DEFAULT_PROGRESS_SPINNER,
+            "Downloading and searching metadata from Azure".into(),
+        );
         let popgetter = Popgetter::new_with_config(config).await?;
+        let search_results = popgetter.search(self.search_params_args.clone().into());
 
-        let mut data = popgetter
-            .search(self.search_params_args.clone().into())
-            .download(&popgetter.config)
-            .await?;
+        sp.stop();
+
```

Comment:
I think feature wise it would be nice to print the number of results here so that the user can mash ctrl-c if they see it trying to download 200,000 metrics while connected to mobile hotspot :)

```suggestion
        let len_requests = search_results.0.shape().0;
        println!("Found {} metrics.", len_requests);
```

Yet another upgrade on this might actually be to print this and prompt for confirmation that the user indeed wants to download this. We could add a `-y` command line option to `DataCommand` which, if passed, will skip the prompt. GitHub doesn't let me comment on lines that are not changed by the PR but the command-line args would be the section which begins with `pub struct DataCommand` in this file.

Reply:
Ah cool, I've never seen this "Add suggestion to batch" option either. Think I'll stick with the "commit suggestion" and have a go with the command line bit.